packages/repository/src/transaction.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Callback} from './common-types';
+
+/**
+ * Local transaction
+ */
+export interface Transaction {
+  /**
+   * Commit the transaction
+   * @param callback
+   */
+  commit(callback?: Callback<any>): void;
```

Comment:
`commit(callback?: Callback<any>): ValueOrPromise<void>`?

Reply:
Let's not introduce callbacks to LB4 APIs please. I believe we have already added support for promises to `Transaction` members in loopback-connector - see https://github.com/strongloop/loopback-connector/pull/147.

IMO, we should use the following form:

```suggestion
  commit(): Promise<void>;
```

Regarding `commit(callback?: Callback<any>): ValueOrPromise<void>`: this does not offer enough type safety. It allows callers to provide a callback and await the returned value at the same time. Here is a more correct type definition:

```ts
commit(callback: Callback<any>): void;
commit(): Promise<void>;
```

Reply:
Since we just want Promise style support in LB4, I've only included `commit(): Promise<void>;` signature for the methods.